stl/inc/yvals_core.h

```diff
@@ -584,7 +584,7 @@
 #define _MSVC_STL_UPDATE  202112L
 
 #ifndef _ALLOW_COMPILER_AND_STL_VERSION_MISMATCH
-#ifdef __CUDACC__
+#if defined(__CUDACC__) && !defined(__clang__)

```

Comment:
No change requested: I'm happy with this minimal edit. Although the same effect could be achieved by moving the Clang check first, that would have additional side effects (IIRC, there's an EDG-imitating-Clang mode that we don't really test right now, and I don't want to disturb it).

At some point we may want to work through the entire matrix of compilers-imitating-compilers that we support, and define centralized macros for the STL to test, but I don't think that's worth the effort right now.

Reply:
Given that we want to check NVCC version here, checking if `__CUDACC_VER_MAJOR__` is defined would do the job w/o ruling out other compilers.

Reply:
I'm applying @Artem-B's suggestion, given that it also addresses DevCom-1572308[^1].
```suggestion
#if defined(__CUDACC__) && defined(__CUDACC_VER_MAJOR__)
```
[^1]: CUDA has msbuild integration that uses `cl /showIncludes /D__CUDACC__` to gather dependencies. The existing `__CUDACC__` check breaks this, but it will nicely slip through with the proposed change.